aii-opennebula/src/main/perl/opennebula.pm

```diff
@@ -170,48 +177,92 @@ sub new
         push(@qimages, $imagename);
-        my @existimage = $one->get_images(qr{^$imagename$});
-        foreach my $t (@existimage) {
-            if (($t->{extended_data}->{TEMPLATE}->[0]->{QUATTOR}->[0]) && ($forcecreateimage)) {
-                # It's safe, we can remove the image
-                $main::this_app->info("Removing VM image: $imagename");
-                my $id = $t->delete();
-                if ($id) {
-                    push(@rimages, $imagename);
-                } else {
-                    $main::this_app->error("VM image: $imagename was not removed");
-                }
-            } else {
-                $main::this_app->info("No QUATTOR flag found for VM image: $imagename");
-            }
-        }
-    	# And create the new image with the image data
-        if (!$remove) {
-            my $newimage = $one->create_image($imagedata->{image}, $imagedata->{datastore});
-            if ($newimage) {
-                $main::this_app->info("Created new VM image ID: ", $newimage->id);
-                push(@nimages, $imagename);
-            } else {
-                $main::this_app->error("VM image: $imagename is not available");
-            }
-    	}
+        $self->remove_vm_images($one, $forcecreateimage, $imagename, \@rimages);
+        # And create the new image with the image data
+        $self->create_vm_images($one, $imagename, $imagedata, $remove, \@nimages);
     }
-
+    # Check created/removed image lists
     if ($remove) {
         my $diff = $self->check_vm_images_list(\@rimages, \@qimages);
         if ($diff) { 
-            $main::this_app->error("Removing these VM images: ", join(',', @qimages));
+            $main::this_app->error("Removing these VM images: ", join(', ', @qimages));
         }
     } else {
         my $diff = $self->check_vm_images_list(\@nimages, \@qimages);
         if ($diff) { 
-            $main::this_app->error("Creating these VM images: ", join(',', @qimages));
+            $main::this_app->error("Creating these VM images: ", join(', ', @qimages));
+        }
+    }
+}
+
+# Create new VM images
+# Returns new images array
+sub create_vm_images
+{
+    my ($self, $one, $imagename, $imagedata, $remove, $ref_nimages) = @_;
+    my $newimage;
+    if (!$remove) {
+        if ($self->is_one_resource_available($one, "image", $imagename)) {
+            $main::this_app->error("Image: $imagename is already used and AII hook: image is not set. ",
+                                    "Please remove this image first.");
+        } else {
+            $newimage = $one->create_image($imagedata->{image}, $imagedata->{datastore});
         }
+        if ($newimage) {
+            $main::this_app->info("Created new VM image ID: ", $newimage->id);
+            push(@{$ref_nimages}, $imagename);
+        } else {
+            $main::this_app->error("VM image: $imagename is not available");
+        }
+    }
+}
+
+# Removes current VM images
+# Returns removed images array
```

Comment:
@piojo you are right, the comment is outdated, now create_image and remove_image functions do not return any value.